### DISCUSSION TITLE
Where is the test.py file on GitHub?

### DISCUSSION THREAD

**Discussion by akhilpaluri2022** [ultralytics/yolov5#5758] [Q&A]
Hey, it looks in the https://docs.ultralytics.com/tutorials/train-custom-datasets/, at the very bottom of the page, it talks about having the 4 .py files. Test.py does not exist in the GitHub. Where can I find this test.py file?

**Comment by zhiqwang** [comment]
I guess it was renamed to https://github.com/ultralytics/yolov5/blob/master/val.py now :)

**Comment by akhilpaluri2022** [comment]
Hi Zhiqiang Wang,

Is the val.py file the same as test.py file?

Thanks,

Akhil Paluri

**Reply by glenn-jocher** [reply]
yes

**Comment by akhilpaluri2022** [comment]
Hi Glenn,

Where is the test.py file?

Thanks,

Akhil Paluri

**Reply by glenn-jocher** [reply]
test.py is renamed as val.py

**Comment by akhilpaluri2022** [comment]
Hi Glenn,

Can you send the test.py and val.py file to me in email?

Thanks,

Akhil Paluri

**Reply by glenn-jocher** [reply]
We don't provide email support for community users.

**Comment by akhilpaluri2022** [comment]
Hi Glenn,

Can you please let me know why you don't provide email support
for community users?

Thanks,

Akhil Palur

**Comment by akhilpaluri2022** [comment]
Hi Glenn,

Can you send the test.py and val.py file to me in email?

Thanks,

Akhil Paluri

**Comment by akhilpaluri2022** [comment]
Hi Glenn,

If the val.py file is the same as test.py file, then can you please let me
know what I need to change in the test.py file to make test.py file
complete?

Thanks,

Akhil Paluri

**Reply by isspid** [reply]
What exactly do you want to do?

**Comment by akhilpaluri2022** [comment]
Hi Isspid,

I am doing a project for my Senior Design Capstone class. My part of the
project is Image Processing. I am using YoloV5 and Pytorch to train my
model. When I installed YoloV5 and Pytorch, there were a few Python files
such as detect.py, export.py, hubconf.py, train.py, and val.py. I found a
website on how to train custom data on Python. At the bottom of the page,
it has 4 Python files such as train.py, test.py, detect.py, and export.py.
test.py does not exist in the file explorer with all of the other files.
When I clicked on test.py on the website, it takes me to the GitHub page
saying that the file was not found.

Where can I find this file?

Thanks,

Akhil Paluri

**Reply by isspid** [reply]
I don't think you have to use that file precisely. If you have used YOLOv5 to train your model, I think you can save the weights of your trained model and just `git pull` or simply clone this repository from scratch with `git pull https://github.com/ultralytics/yolov5`. After having cloned the correct repository, you can then use the file called `val.py` to get results for your model. I hope I was clear.

**Comment by akhilpaluri2022** [comment]
Hi Isspid,

Can you please let me know what you typed in google search to get the
GitHub link above?

Thanks,

Akhil Paluri

**Comment by akhilpaluri2022** [comment]
Hi Isspid,

I found where you found that link on google.

Can you please let me know what I should be doing from what you mentioned
in the previous email?

Thanks,

Akhil Paluri

**Reply by isspid** [reply]
You should follow [this tutorial](https://docs.ultralytics.com/tutorials/train-custom-datasets/) step by step and I am sure it will work. Even if you would have to train your model again, the training shouldn't take very long.

**Comment by akhilpaluri2022** [comment]
Hi Isspid,

The link that you put above is what I am using. I am having trouble in
figuring out what you mentioned in your bg email and also making sure that
the paths are correct in my yaml file.

Thanks,

Akhil Paluri

**Reply by isspid** [reply]
What is your problem exactly? Provide some screenshots or some more information. Your comments are really vague.

**Comment by akhilpaluri2022** [comment]
Hi Isspid,

Can you please let me know if we can do a zoom meeting to discuss this?

Thanks,

Akhil Paluri
